etna/core/middleware.py

```diff
@@ -122,3 +125,53 @@ def process_template_response(
             ),
         )
         return response
+
+
+class SearchMiddleware:
+    """
+    Handles the session settings for "back to search results" functionality in the record details page.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        return response
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        # create or retain session var
+        if hasattr(view_func, "view_class"):
+            # set session when navigating from search view containing links to record details page
+            if view_func.view_class in (CatalogueSearchView, FeaturedSearchView):
+                request.session.set_expiry(settings.SESSION_EXPIRY_VALUE)
+                request.session["back_to_search_url"] = request.get_full_path()
+                return None
```

Comment:
Why do we need a Middleware for this? Why not do this in the views themselves?

Reply:
I thought to have it in one place - create+delete the sessions. Its not needed here otherwise. I've moved it to the views.